pallets/file-system/src/utils.rs

```diff
@@ -159,7 +161,32 @@ where
             Error::<T>::BspAlreadyVolunteered
         );
 
-        // TODO: Check that the BSP XOR is lower than the threshold
+        // Compute BSP's threshold
+        let bsp_threshold = Self::compute_bsp_xor(
+            fingerprint
+                .as_ref()
+                .try_into()
+                .map_err(|_| Error::<T>::FailedToEncodeFingerprint)?,
+            &who.encode()
+                .try_into()
+                .map_err(|_| Error::<T>::FailedToEncodeBsp)?,
+        )?;
+
+        // Get number of blocks since the storage request was issued.
+        let blocks_since_requested = <frame_system::Pallet<T>>::block_number()
+            .saturating_sub(file_metadata.requested_at)
+            .saturated_into::<u32>();
```

Comment:
Why as `u32`?

Reply:
ya, caught this as well. It is not `.try_into()` from block number to `ThresholdType`